src/Common/Common.Api/Reports/ReportsClient.cs

```diff
@@ -17,7 +19,79 @@ public ReportsClient(IPowerBIClient client) : base(client)
 
         public IEnumerable<Report> GetReports()
         {
-            return this.Client.Reports.GetReports().Value.Select(x => (Report)x);
+            return this.Client.Reports.GetReports().Value?.Select(x => (Report)x);
+        }
+
+        public IEnumerable<Report> GetReportsAsAdmin(string filter = null, int? top = null, int? skip = null)
+        {
+            return this.Client.Reports.GetReportsAsAdmin(filter: filter, top: top, skip: skip).Value?.Select(x => (Report)x);
+        }
+
+        public IEnumerable<Report> GetReportsAsAdminForWorkspace(Guid workspaceId, string filter = null, int? top = null, int? skip = null)
+        {
+            return this.Client.Reports.GetReportsAsAdmin(groupId: workspaceId.ToString(), filter: filter, top: top, skip: skip).Value?.Select(x => (Report)x);
+        }
+
+        public IEnumerable<Report> GetReportsForWorkspace(Guid workspaceId)
+        {
+            return this.Client.Reports.GetReports(groupId: workspaceId.ToString()).Value?.Select(x => (Report)x);
+        }
+
+        public Stream ExportReport(Guid reportId, Guid? workspaceId = default)
+        {
+            return workspaceId.HasValue && workspaceId.Value != default ?
+                this.Client.Reports.ExportReport(groupId: workspaceId.Value.ToString(), reportKey: reportId.ToString()) :
+                this.Client.Reports.ExportReport(reportKey: reportId.ToString());
+        }
+
+        public IEnumerable<Dashboard> GetDashboards()
+        {
+            return this.Client.Dashboards.GetDashboards().Value?.Select(x => (Dashboard)x);
+        }
+
+        public IEnumerable<Dashboard> GetDashboardsAsAdmin(string filter = default, int? top = default, int? skip = default)
+        {
+            return this.Client.Dashboards.GetDashboardsAsAdmin(filter: filter, top: top, skip: skip).Value?.Select(x => (Dashboard)x);
+        }
+
+        public IEnumerable<Dashboard> GetDashboardsForWorkspace(Guid workspaceId)
+        {
+            return this.Client.Dashboards.GetDashboards(groupId: workspaceId.ToString()).Value?.Select(x => (Dashboard)x);
+        }
+
+        public IEnumerable<Dashboard> GetDashboardsAsAdminForWorkspace(Guid workspaceId, string filter = default, int? top = default, int? skip = default)
```

Comment:
For these "ForWorkspace" wrapper methods, I personally like having "AsAdmin" as the end.

Reply:
We can change that

Reply:
Added as part of #35 too